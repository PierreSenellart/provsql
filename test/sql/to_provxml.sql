\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE result_provxml AS SELECT
  *, to_provxml(provenance(),'personnel_name') AS provxml FROM (
    (SELECT DISTINCT city FROM personnel)
  EXCEPT
    (SELECT p1.city
     FROM personnel p1, personnel p2
     WHERE p1.city = p2.city AND p1.id < p2.id
     GROUP BY p1.city)
  ) t
ORDER BY city;

SELECT remove_provenance('result_provxml');
SELECT city, regexp_replace(provxml,'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}','UUID','g') AS provxml FROM result_provxml;

DROP TABLE result_provxml;

CREATE TABLE result_provxml AS SELECT
  *, to_provxml(provenance()) AS provxml FROM (
    (SELECT DISTINCT city FROM personnel)
  EXCEPT
    (SELECT p1.city
     FROM personnel p1, personnel p2
     WHERE p1.city = p2.city AND p1.id < p2.id
     GROUP BY p1.city)
  ) t
ORDER BY city;

SELECT remove_provenance('result_provxml');
SELECT city, regexp_replace(provxml,'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}','UUID','g') AS provxml FROM result_provxml;

DROP TABLE result_provxml;
