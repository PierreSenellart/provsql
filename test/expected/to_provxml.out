\set ECHO none
remove_provenance

(1 row)
city|provxml
Berlin|<?xml version='1.0' encoding='UTF-8'?>
<prov:document xmlns:prov='http://www.w3.org/ns/prov#' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:provsql='https://provsql.org/'>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:monus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>left</prov:label>
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>right</prov:label>
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>Ellen</prov:value>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>Susan</prov:value>
  </prov:entity>
</prov:document>

New York|<?xml version='1.0' encoding='UTF-8'?>
<prov:document xmlns:prov='http://www.w3.org/ns/prov#' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:provsql='https://provsql.org/'>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:monus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>left</prov:label>
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>right</prov:label>
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>John</prov:value>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>Paul</prov:value>
  </prov:entity>
</prov:document>

Paris|<?xml version='1.0' encoding='UTF-8'?>
<prov:document xmlns:prov='http://www.w3.org/ns/prov#' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:provsql='https://provsql.org/'>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:monus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>left</prov:label>
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>right</prov:label>
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>Dave</prov:value>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>Magdalen</prov:value>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
    <prov:value>Nancy</prov:value>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
</prov:document>

(3 rows)
remove_provenance

(1 row)
city|provxml
Berlin|<?xml version='1.0' encoding='UTF-8'?>
<prov:document xmlns:prov='http://www.w3.org/ns/prov#' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:provsql='https://provsql.org/'>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:monus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>left</prov:label>
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>right</prov:label>
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
</prov:document>

New York|<?xml version='1.0' encoding='UTF-8'?>
<prov:document xmlns:prov='http://www.w3.org/ns/prov#' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:provsql='https://provsql.org/'>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:monus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>left</prov:label>
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>right</prov:label>
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
</prov:document>

Paris|<?xml version='1.0' encoding='UTF-8'?>
<prov:document xmlns:prov='http://www.w3.org/ns/prov#' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:provsql='https://provsql.org/'>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:monus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>left</prov:label>
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
    <prov:label>right</prov:label>
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:plus</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:input</prov:type>
  </prov:entity>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:entity prov:id='provsql:UUID'>
    <prov:type xsi:type='xsd:QName'>provsql:times</prov:type>
  </prov:entity>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
  <prov:wasDerivedFrom>
    <prov:generatedEntity prov:ref='provsql:UUID' />
    <prov:usedEntity prov:ref='provsql:UUID' />
  </prov:wasDerivedFrom>
</prov:document>

(3 rows)
